File: backend/app/models/comment.py
from datetime import datetime

from sqlalchemy import Column, DateTime, ForeignKey, Integer, Text
from sqlalchemy.orm import relationship

from app.db.database import Base


class Comment(Base):
    __tablename__ = "comments"

    id = Column(
        Integer,
        primary_key=True,
        index=True,
    )

    issue_id = Column(
        Integer,
        ForeignKey("issues.id", ondelete="CASCADE"),
        nullable=False,
    )

    user_id = Column(
        Integer,
        ForeignKey("users.id"),
        nullable=False,
    )

    content = Column(
        Text,
        nullable=False,
    )

    created_at = Column(
        DateTime,
        default=datetime.utcnow,
        nullable=False,
    )

    updated_at = Column(
        DateTime,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
        nullable=False,
    )

    issue = relationship(
        "Issue",
        back_populates="comments",
    )

    user = relationship(
        "User",
        back_populates="comments",
    )

File: frontend/src/types/comment.ts
// GET /issues/{issue_id}/comments
export interface Comment {
  id: number;
  issue_id: number;
  user_id: number;
  content: string;
  created_at: string;
  updated_at: string;
}

// POST /issues/{issue_id}/comments
export interface CreateCommentRequest {
  content: string;
}

// PATCH /comments/{comment_id}
export interface UpdateCommentRequest {
  content: string;
}